src/components/ConfigureSupersetEmbeddedDashboardModal/UpdateSupersetEmbeddedDashboard.js

```diff
@@ -0,0 +1,219 @@
+                showChartControls:
+                    !dashboard.embedded.options.hideChartControls,
+                showFilters: dashboard.embedded.options.filters.visible,
+            })
+        }
+    }, [dashboard, resetStateWithNewValues])
+
+    return (
+        <Modal>
+            <form onSubmit={handleSubmit}>
+                <ModalTitle>
+                    {i18n.t('Edit external dashboard', { nsSeparator: '###' })}
+                </ModalTitle>
+                <ModalContent className={styles.modalContent}>
+                    {(queryLoading || !!queryError) && (
+                        <div
+                            className={cx(styles.contentOverlay, {
+                                [styles.loading]: queryLoading,
+                                [styles.error]: queryError,
+                            })}
+                        >
+                            {queryLoading && <CircularLoader />}
+                            {queryError && (
+                                <NoticeBox
+                                    error
+                                    title={i18n.t(
+                                        'Could not load dashboard details'
+                                    )}
+                                >
+                                    {parseErrorText(queryError)}
+                                </NoticeBox>
+                            )}
+                        </div>
+                    )}
+                    <SupersetEmbeddedDashboardFields
+                        isSupersetEmbedIdValid={isSupersetEmbedIdValid}
+                        isSupersetEmbedIdFieldTouched={
+                            isSupersetEmbedIdFieldTouched
+                        }
+                        values={values}
+                        onChange={onChange}
+                        onSupersetEmbedIdFieldBlur={onSupersetEmbedIdFieldBlur}
+                        submitting={mutationLoading || queryLoading}
+                    />
+                    {mutationError && (
+                        <NoticeBox error title={mutationErrorTitle}>
+                            {mutationErrorText}
+                        </NoticeBox>
+                    )}
+                </ModalContent>
+                <ModalActions>
+                    <div className={styles.buttonStrip}>
+                        {!queryError && (
+                            <Button
+                                loading={mutationLoading}
+                                type="submit"
+                                primary
+                                disabled={
+                                    !hasFieldChanges ||
+                                    !isSupersetEmbedIdValid ||
+                                    queryLoading
+                                }
+                            >
+                                {i18n.t('Update dashboard')}
+                            </Button>
+                        )}
```

Comment:
Note that I am not checking on `dashboard.access.edit` here, or checking for small screens. this is because this modal only pops up after clicking the EDIT button in the dashboards-bar which already checks these things.